Composer/packages/client/src/store/reducer/index.ts

```diff
@@ -435,12 +446,39 @@ const publishFailure: ReducerFunc = (state, { error }) => {
 };
 
 const getPublishStatus: ReducerFunc = (state, payload) => {
-  if (payload.results?.botStatus === 'connected') {
+  // the action below only applies to when a bot is being started using the "start bot" button
+  // a check should be added to this that ensures this ONLY applies to the "default" profile.
+  if (payload.target.name === 'default' && payload.endpointURL) {
     state.botStatus = BotStatus.connected;
+    state.botEndpoints[state.projectId] = `${payload.endpointURL || 'http://localhost:3979'}/api/messages`;
+  }
+
+  // if no history exists, create one with the latest status
+  // otherwise, replace the latest publish history with this one
+  if (!state.publishHistory[payload.target.name] && payload.status !== 404) {
+    state.publishHistory[payload.target.name] = [payload];
+  } else if (payload.status !== 404) {
+    // TODO: this should only happen if they actually represent the same item...
+    state.publishHistory[payload.target.name][0] = payload;
   }
   return state;
 };
 
+const getPublishStatusFailed: ReducerFunc = (state, payload) => {
```

Comment:
This looks like the same code from L458. Can you combine?

Reply:
Combined